lib/knife/changelog/changelog.rb

```diff
@@ -100,28 +100,34 @@ def clean
 
     def handle_new_cookbook
       stars = '**' if @config[:markdown]
-      ["#{stars}Cookbook was not in the berksfile#{stars}"]
+      ["#{stars}Cookbook was not in the berksfile/policyfile#{stars}"]
     end
 
     def execute(name, submodule = false)
-      version_change, changelog = if submodule
-                                    handle_submodule(name)
-                                  elsif new_cookbook?(name)
-                                    ['', handle_new_cookbook]
-                                  else
-                                    case true
-                                    when supermarket?(name)
-                                      handle_source(name)
-                                    when git?(name)
-                                      handle_git(name, git_location(name))
-                                    when local?(name)
-                                      Chef::Log.debug "path location are always at the last version"
-                                      ['', '']
+      begin
+        version_change, changelog = if submodule
+                                      handle_submodule(name)
+                                    elsif new_cookbook?(name)
+                                      ['', handle_new_cookbook]
                                     else
-                                      raise "Cannot handle #{loc.class} yet"
+                                      case true
+                                      when git?(name)
+                                        handle_git(name, git_location(name))
+                                      when supermarket?(name)
+                                        handle_source(name)
+                                      when local?(name)
+                                        Chef::Log.debug "path location are always at the last version"
+                                        ['', '']
+                                      else
+                                        raise "Cannot handle #{loc.class} yet"
+                                      end
                                     end
-                                  end
-      format_changelog(name, version_change, changelog)
+      rescue StandardError
+        raise unless @config[:allow_errors]
```

Comment:
Why not just display the error message and forget about the new flag?

Reply:
See previous comment

Reply:
ack